### ISSUE TITLE
Merge packages and extensions

### ISSUE THREAD

**Issue by syl20bnr** [syl20bnr/spacemacs#5]
This task depends on #110 because we need to retain backward compatibility with the current layers format.
The idea is to have only one list per configuration layer to declare all the packages and extensions.
This list will have convenient properties like:
- `:type` with possible values `melpa`, `melpa-stable`, `el-get` or `quelpa` or even `submodule` (although we definitely don't want them).
- `:step` with values `pre` or `post` to control the loading time.
- `:disabled t` to mark a package as not activated by default (useful to be able to incorporate contrib layers with only one package in spacemacs layer).

**Comment by syl20bnr** [comment]
Note to self: check el-get recipes

**Comment by tuhdo** [comment]
@syl20bnr `quelpa` is merged into MELPA: https://github.com/milkypostman/melpa/pull/2968

Maybe it is the time?

**Comment by syl20bnr** [comment]
Almost one year later it is now implemented in develop!
